### PR TITLE
fix(gomod): Use modcacherw flag starting from 1.14

### PR DESCRIPTION
**Pull request by zharinov**
## Context:

Ref: #9748

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
I'm unable to reproduce the artifact error, so can't tell if it works well or not

File: lib/manager/gomod/artifacts.ts [hunk 1]
```diff
@@ -68,6 +69,20 @@ function getUpdateImportPathCmds(
   return updateImportCommands;
 }
 
+function useModcacherw(goVersion: string): boolean {
+  if (!is.string(goVersion)) {
+    return false;
```
Comment (rarkins) [inline lib/manager/gomod/artifacts.ts:74]: If no constraint then we run the latest, so maybe return `true` here?

File: lib/manager/gomod/artifacts.ts [hunk 2]
```diff
@@ -68,6 +69,20 @@ function getUpdateImportPathCmds(
...
+  }
+
+  const [majorPart, minorPart] = goVersion.replace(/^[^\d]*/g, '').split('.');
+  const [major, minor] = [majorPart, minorPart].map((x) => parseInt(x, 10));
+  if (Number.isNaN(major) || Number.isNaN(minor)) {
+    return false;
+  }
+
+  return major > 1 || (major === 1 && minor >= 14);
```
Comment (rarkins) [inline lib/manager/gomod/artifacts.ts:83]: Can you make sure it would work if one day they specify the constraint using `major.minor.patch`?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/manager/gomod/artifacts.ts (hunk 2 above)
Comment (zharinov) [reply lib/manager/gomod/artifacts.ts]: I've done double-check, it just ignores patch

File: lib/manager/gomod/artifacts.ts [hunk 3]
```diff
@@ -68,6 +69,21 @@ function getUpdateImportPathCmds(
   return updateImportCommands;
 }
 
+function useModcacherw(goVersion: string): boolean {
+  if (!is.string(goVersion)) {
+    return true;
+  }
+
+  const [majorPart, minorPart] = goVersion.replace(/^[^\d]*/g, '').split('.');
```
Comment (viceice) [inline lib/manager/gomod/artifacts.ts:77]: `/^[^\d]*/g` won't this remove all dots? Isn't this simpler?
```suggestion
  const [,majorPart, minorPart] = goVersion.match(/(\d+)\.(\d+)/) ?? [];
```

**Comment by clok** [comment]
I noticed [this fork](https://github.com/renovate-testing/test-9748-ghlabels/pull/2) with working PRs of the [ghlabels](https://github.com/clok/ghlabels/pull/3) project. Is this change related to resolving the patching issues?
https://github.com/renovate-testing/test-9748-ghlabels/pull/2

**Review by rarkins — APPROVED** [review]

**Comment by clok** [comment]
Confirmed that this is working as expected on my projects. Thank you for the fix!

**Pull request merged**